exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -187,6 +187,7 @@ def send_bytes(bytes, timeout:) # rubocop:disable Metrics/AbcSize, Metrics/Cyclo
             when Net::HTTPBadRequest, Net::HTTPClientError, Net::HTTPServerError
               # TODO: decode the body as a google.rpc.Status Protobuf-encoded message when https://github.com/open-telemetry/opentelemetry-collector/issues/1357 is fixed.
               response.body # Read and discard body
+              @metrics_reporter.add_to_counter('otel.otlp_exporter.failure', labels: { 'reason' => response.code })
```

Comment:
Currently this failure was not being reported so comparing `otel.bsp.export.failure` against `otel.otlp_exporter.failure` was not providing proper insight into export failures.

Reply:
✨

Reply:
@robertlaurin Is `reason` free text other instances? Would it be sensible to use `http.status_code` as a label instead?

Reply:
I'd consider it free text at the moment
https://github.com/open-telemetry/opentelemetry-ruby/blob/469aa89ba180b1716a6a33c12647b32f1df4fed9/exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb#L200-L214